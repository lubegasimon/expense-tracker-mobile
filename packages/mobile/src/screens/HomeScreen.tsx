import React from "react";
import { Button, View } from "react-native";
import { useNavigation } from "@react-navigation/native";
import { StackNavigationProp } from "@react-navigation/stack";
import { RouteNavigationStack } from "@/src/types";

type SignupNavigationProp = StackNavigationProp<RouteNavigationStack, "signup">;

export default function HomeScreen() {
  const navigation = useNavigation<SignupNavigationProp>();
  return (
    <View>
      <Button
        title="Create your account"
        onPress={() => navigation.navigate("signup")}
      />
    </View>
  );
}
