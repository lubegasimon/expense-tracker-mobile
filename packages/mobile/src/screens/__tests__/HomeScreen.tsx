import { render, screen } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";

import HomeScreen from "../HomeScreen";

describe("<HomeScreen/>", () => {
  test("HomeScreen contains Button", () => {
    const tree = render(
      <NavigationContainer>
        <HomeScreen />
      </NavigationContainer>,
    );
    expect(tree).toMatchSnapshot();
    screen.getByRole("button", { name: "Create your account" });
  });
});
