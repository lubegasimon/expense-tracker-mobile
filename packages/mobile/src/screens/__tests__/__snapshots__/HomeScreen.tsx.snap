// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen/> HomeScreen contains Button 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Create your account
      </Text>
    </View>
  </View>
</View>
`;